File: server/route/admin/subject.py
from fastapi import APIRouter
from sqlalchemy.ext.asyncio import AsyncSession
from store.database import GetDB
from service.admin import SubjectService
from schema.common import ResponseModel, SimpleNameParams


router = APIRouter(prefix="/subject")


@router.get("/all")
async def search(db: AsyncSession = GetDB):
    res = await SubjectService.all(db)
    return ResponseModel(data=res)


@router.post("/")
async def create(params: SimpleNameParams, db: AsyncSession = GetDB):
    id = await SubjectService.create(db, params.name.strip())
    return ResponseModel(data=id)


@router.patch("/{id}")
async def update(id: str, params: SimpleNameParams, db: AsyncSession = GetDB):
    await SubjectService.update(db, id, params.name.strip())
    return ResponseModel()


@router.delete("/{id}")
async def delete(id: str, db: AsyncSession = GetDB):
    await SubjectService.delete(db, id)
    return ResponseModel()

File: admin/src/services/subject.ts
import { request } from '@umijs/max';

export const SubjectApi = {
  list: async () => {
    const res = await request<ApiData<Subject[]>>('/subject/all');
    return res.data;
  },

  actives: async () => {
    const res = await request<ApiData<Subject[]>>('/subject/actives');
    return res.data;
  },

  create: async (data: { name: string }) => {
    const res = await request<ApiData<Subject>>('/subject', {
      method: 'POST',
      data,
    });
    return res.status === 0;
  },

  update: async (id: number, data: { name: string }) => {
    const res = await request<ApiData<Subject>>(`/subject/${id}`, {
      method: 'PATCH',
      data,
    });
    return res.status === 0;
  },

  delete: async (id: number) => {
    const res = await request<ApiData<void>>(`/subject/${id}`, {
      method: 'DELETE',
    });
    return res.status === 0;
  },

  toggleStatus: async (id: number, status: number) => {
    const res = await request<ApiData<Subject>>(`/subject/${id}/status`, {
      method: 'PUT',
      data: { status },
    });
    return res.status === 0;
  },
};
